SwrveSDK/src/amazon/java/com/swrve/sdk/adm/SwrveAdmIntentService.java

```diff
@@ -0,0 +1,254 @@
+package com.swrve.sdk.adm;
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.ContentResolver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.NotificationCompat;
+import android.support.v4.content.ContextCompat;
+
+import com.amazon.device.messaging.ADMMessageHandlerBase;
+import com.google.gson.reflect.TypeToken;
+import com.swrve.sdk.ISwrveBase;
+import com.swrve.sdk.Swrve;
+import com.swrve.sdk.SwrveHelper;
+import com.swrve.sdk.SwrveLogger;
+import com.swrve.sdk.SwrvePushEngageReceiver;
+import com.swrve.sdk.SwrveSDK;
+import com.swrve.sdk.qa.SwrveQAUser;
+
+import com.google.gson.Gson;
+
+import java.util.Date;
+import java.util.Iterator;
+import java.util.LinkedList;
+
+public class SwrveAdmIntentService extends ADMMessageHandlerBase {
+    private final static String TAG = "SwrveAdm";
+    private final static String AMAZON_RECENT_PUSH_IDS = "recent_push_ids";
+    private final static String AMAZON_PREFERENCES = "swrve_amazon_pref";
+    private final int MAX_ID_CACHE_ITEMS = 16;
```

Comment:
Going with SwrveAdmConstants.PUSH_ID_CACHE_SIZE_KEY = "_s.c" - check if it exists, default to DEFAULT_PUSH_ID_CACHE_SIZE (16) if not. Sanity check on final value to make sure it's at least 0.